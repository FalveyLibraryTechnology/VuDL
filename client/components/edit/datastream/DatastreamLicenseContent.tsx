import React, { useEffect, useState } from "react";
import Button from "@mui/material/Button";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import FormControl from "@mui/material/FormControl";
import FormControlLabel from "@mui/material/FormControlLabel";
import FormLabel from "@mui/material/FormLabel";
import Radio from "@mui/material/Radio";
import RadioGroup from "@mui/material/RadioGroup";
import useDatastreamOperation from "../../../hooks/useDatastreamOperation";
import { useEditorContext } from "../../../context/EditorContext";

const DatastreamLicenseContent = (): React.ReactElement => {
    const {
        state: { licensesCatalog },
        action: { toggleDatastreamModal },
    } = useEditorContext();
    const { uploadLicense, getLicenseKey } = useDatastreamOperation();
    const [licenseKey, setLicenseKey] = useState("");
    useEffect(() => {
        const getLicense = async () => {
            setLicenseKey(await getLicenseKey());
        };
        getLicense();
    }, []);
    return (
        <>
            <DialogContent>
                <FormControl>
                    <FormLabel>Choose License</FormLabel>
                    <RadioGroup value={licenseKey} onChange={(event) => setLicenseKey(event.target.value)}>
                        {Object.entries(licensesCatalog).map(([key, license]) => {
                            return (
                                <FormControlLabel
                                    key={key}
                                    value={key}
                                    control={<Radio />}
                                    label={license.name}
                                    sx={{ margin: "0" }}
                                />
                            );
                        })}
                    </RadioGroup>
                </FormControl>
            </DialogContent>

            <DialogActions>
                <Button
                    className="uploadLicenseButton"
                    onClick={async () => {
                        await uploadLicense(licenseKey);
                    }}
                >
                    Save
                </Button>
                <Button onClick={toggleDatastreamModal}>Cancel</Button>
            </DialogActions>
        </>
    );
};

export default DatastreamLicenseContent;
