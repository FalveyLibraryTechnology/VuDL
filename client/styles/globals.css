:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
