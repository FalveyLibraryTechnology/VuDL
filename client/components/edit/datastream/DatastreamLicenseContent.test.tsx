import React from "react";
import { describe, beforeEach, expect, it, jest } from "@jest/globals";
import { mount, shallow } from "enzyme";
import { act } from "react-dom/test-utils";
import toJson from "enzyme-to-json";
import DatastreamLicenseContent from "./DatastreamLicenseContent";

jest.mock("@mui/material/RadioGroup", () => () => "RadioGroup");
const mockUseEditorContext = jest.fn();
jest.mock("../../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));
const mockUseDatastreamOperation = jest.fn();
jest.mock("../../../hooks/useDatastreamOperation", () => () => {
    return mockUseDatastreamOperation();
});

describe("DatastreamLicenseContent", () => {
    let editorValues;
    let datastreamOperationValues;
    beforeEach(() => {
        editorValues = {
            state: {
                licensesCatalog: {
                    testLicenseKey: {
                        name: "testLicense",
                    },
                },
            },
            action: {
                toggleDatastreamModal: jest.fn(),
            },
        };
        datastreamOperationValues = {
            uploadLicense: jest.fn(),
            getLicenseKey: jest.fn(),
        };
        mockUseEditorContext.mockReturnValue(editorValues);
        mockUseDatastreamOperation.mockReturnValue(datastreamOperationValues);
    });

    it("renders", () => {
        const wrapper = shallow(<DatastreamLicenseContent />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("calls uploadLicense on click", async () => {
        datastreamOperationValues.uploadLicense.mockResolvedValue("upload worked");
        const wrapper = mount(<DatastreamLicenseContent />);

        await act(async () => {
            wrapper.find("button.uploadLicenseButton").simulate("click");
            wrapper.update();
        });
        expect(datastreamOperationValues.getLicenseKey).toHaveBeenCalled();
        expect(datastreamOperationValues.uploadLicense).toHaveBeenCalled();
    });
});
