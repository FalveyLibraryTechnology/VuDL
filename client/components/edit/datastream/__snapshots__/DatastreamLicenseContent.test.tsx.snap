// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatastreamLicenseContent renders 1`] = `
<Fragment>
  <ForwardRef(DialogContent)>
    <ForwardRef(FormControl)>
      <ForwardRef(FormLabel)>
        Choose License
      </ForwardRef(FormLabel)>
      <Component
        onChange={[Function]}
        value=""
      >
        <ForwardRef(FormControlLabel)
          control={<ForwardRef(Radio) />}
          key="testLicenseKey"
          label="testLicense"
          sx={
            {
              "margin": "0",
            }
          }
          value="testLicenseKey"
        />
      </Component>
    </ForwardRef(FormControl)>
  </ForwardRef(DialogContent)>
  <ForwardRef(DialogActions)>
    <ForwardRef(Button)
      className="uploadLicenseButton"
      onClick={[Function]}
    >
      Save
    </ForwardRef(Button)>
    <ForwardRef(Button)
      onClick={[MockFunction]}
    >
      Cancel
    </ForwardRef(Button)>
  </ForwardRef(DialogActions)>
</Fragment>
`;
